lint-staged.config.js

```diff
@@ -25,7 +25,12 @@ const joinFiles = ( files ) => {
 const PLUGIN_BASE_NAME = path.basename( __dirname );
 
 module.exports = {
-	'**/*.js': ( files ) => `npm run lint-js -- ${ joinFiles( files ) }`,
+	'**/*.{js,ts}': ( files ) => {
+		return [
+			`npm run lint-js -- ${ joinFiles( files ) }`,
+			`npm run tsc`, // TODO: How to pass joinFiles( files ) here?
```

Comment:
I wanted this to rather be:

```suggestion
			`npm run tsc -- ${ joinFiles( files ) }`,
```

But when I do this, I get an error:

```
✖ npm run tsc -- '/home/westonruter/repos/performance/lint-staged.config.js':

> tsc
> tsc /home/westonruter/repos/performance/lint-staged.config.js

error TS6504: File '/home/westonruter/repos/performance/lint-staged.config.js' is a JavaScript file. Did you mean to enable the 'allowJs' option?
  The file is in the program because:
    Root file specified for compilation
husky - pre-commit script failed (code 1)
```

Maybe what I want to do is not supported in TypeScript in the same way that PHPStan needs to run on the entire project and not just the changed files?

Reply:
If this is the case, then:

```suggestion
			`npm run tsc`,
```

Reply:
> But when I do this, I get an error:

`[..] is a JavaScript file. Did you mean to enable the 'allowJs' option?` is pretty clear`.

If you don't want to run tsc against JS files, then don't set up lint-staged + tsc for JS files.

----

Never used tsc with lint-staged, but https://github.com/lint-staged/lint-staged/issues/825 might be helpful. Some people use https://www.npmjs.com/package/tsc-files.

Reply:
Then I think the issue is that the tsconfig file is not read when you pass it a list of files, according to https://github.com/lint-staged/lint-staged/issues/825. That's why some use https://www.npmjs.com/package/tsc-files for that.

Reply:
We could also consider removing tsc from lint-staged for now, probably not worth the hassle. Or just run it for the whole project because it's a tiny project anyway.

Reply:
OK, Let's just run it on the entire project then, same as we do for PHPStan.